stacker/actions/build.py

```diff
@@ -238,26 +264,34 @@ def _launch_stack(self, stack, **kwargs):
         tags = build_stack_tags(stack)
         parameters = self.build_parameters(stack, provider_stack)
 
-        new_status = None
-        if not provider_stack:
-            new_status = SubmittedStatus("creating new stack")
+        if recreate:
+            logger.debug("Re-creating stack: %s", stack.fqn)
+            self.provider.create_stack(stack.fqn, template, parameters,
+                                       tags)
+            return SubmittedStatus("re-creating stack")
+        elif not provider_stack:
             logger.debug("Creating new stack: %s", stack.fqn)
             self.provider.create_stack(stack.fqn, template, parameters,
                                        tags)
-        else:
-            if not should_update(stack):
-                return NotUpdatedStatus()
-            try:
-                new_status = SubmittedStatus("updating existing stack")
+            return SubmittedStatus("creating new stack")
+        elif not should_update(stack):
+            return NotUpdatedStatus()
+
+        try:
+            if self.provider.prepare_stack_for_update(stack.fqn, tags):
                 existing_params = provider_stack.get('Parameters', [])
                 self.provider.update_stack(stack.fqn, template,
                                            existing_params, parameters, tags,
                                            force_interactive=stack.protected)
-                logger.debug("Updating existing stack: %s", stack.fqn)
-            except StackDidNotChange:
-                return DidNotChangeStatus()
 
-        return new_status
+                logger.debug("Updating existing stack: %s", stack.fqn)
+                return SubmittedStatus("updating existing stack")
+            else:
+                logger.debug("Destroying stack for recreation: %s",
```

Comment:
This doesn't seem to actually do any destroying?

Reply:
`prepare_stack_for_update` only returns `False` if the stack will be deleted. The mesage could be placed in a better spot to make that more clear.

Reply:
Yes, at the point the destroying has already been scheduled by `prepare_stack_for_update`. I suppose the message could be clearer and/or better placed.

Reply:
The message was actually redundant, as one was printed at `warn` level in `prepare_stack_for_update`. I removed it.